net: lib: wifi: mgmt: Fix memory leaks

pdgendt opened the pull request:
When connecting to WiFi from stored credentials, the `key_passwd` is never freed.
Additionally if the connect fails, the allocated data was never freed.

pdgendt committed:
net: lib: wifi: mgmt: Fix memory leaks

When connecting to WiFi from stored credentials, the key_passwd is never
freed.
Additionally if the connect fails, the allocated data was never freed.

Convert heap allocated memory to stack allocated buffers.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

jukkar reviewed (commented):
There is now mixed changes in the same commit, could you separate the misra related changes to a separate commit. Also we probably need to backport the changes to 4.2

pdgendt commented:
There were more memory leaks in other (error) cases, I've changed the heap allocated buffers to stack allocated ones. PTAL

jukkar commented on subsys/net/l2/wifi/wifi_mgmt.c:
```diff
@@ -1583,21 +1551,12 @@ static int __stored_creds_to_params(struct wifi_credentials_personal *creds,
-				       creds->header.key_passwd);
-			if (ret > creds->header.key_passwd_length) {
+			if (creds->header.key_passwd_length > WIFI_ENT_PSWD_MAX_LEN) {
 				LOG_ERR("key_passwd string truncated\n");
```
Not your change but the `\n` should not be there as it adds extra newline to log output. Could you fix these as I guess there are more than one of these in this file, in a separate commit?

jukkar reviewed (commented):
Thanks, avoiding malloc here is a good goal.

pdgendt committed:
net: wifi: mgmt: Remove extraneous newlines from logs

Some log statements appended unnecessary newlines, remove those.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

pdgendt committed:
include: zephyr: toolchain: Introduce macros to ignore -Wcast-qual

Allow enabling/disabling the -Wcast-qual warning, for example to cast away
constness.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

pdgendt commented:
Ping @krish2718
